swap_meet/vendor.py

```diff
@@ -1,2 +1,101 @@
+from swap_meet.item import Item
+
+# Wave 1
+"""The class vendor takes in a parameter of inventory. It has a method function
+    that adds and removes items.
+    Returns:
+      It returns the item or the boolean false.
+        """
+
+
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+        self.item = Item()
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+# Wave 2
+        """This method checks through the items csategory and appends it to a list.
+        """
+
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+# Wave 3
+        """This item checks to see if an item is in my inventory or my friends
+        and it removes it from my items to append to theirs. The same is done on
+        their end.
+        """
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        if my_item in self.inventory:
+            self.remove(my_item)
+            vendor.add(my_item)
+        if their_item in vendor.inventory:
+            vendor.remove(their_item)
+            self.add(their_item)
+            return True
+# Wave 4
+        """[summary]
+        """
+
+    def swap_first_item(self, another_vendor):
+        if len(self.inventory) != 0 and len(another_vendor.inventory) != 0:
+            another_vendor.add(self.inventory[0])
+            self.remove(self.inventory[0])
+            self.add(another_vendor.inventory[0])
+            another_vendor.remove(another_vendor.inventory[0])
+            return True
+        else:
+            return False
```

Comment:
A guard clause can also be used here to get rid of the "dangly" else clause. 

```suggestion
        if len(self.inventory) == 0 or len(another_vendor.inventory) == 0:
            return False
        another_vendor.add(self.inventory[0])
        self.remove(self.inventory[0])
        self.add(another_vendor.inventory[0])
        another_vendor.remove(another_vendor.inventory[0])
        return True
```

Reply:
The `remove` method time complexity is O(n). We can improve the performance of this function to O(1) if we did multiple assignment: 

```suggestion
        if len(self.inventory) == 0 or len(another_vendor.inventory) == 0:
            return False
        self.inventory[0], another_vendor.inventory[0] = another_vendor.inventory[0], self.inventory[0]

        return True
```

Here's a resource to test out multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp